main/src/ca/uwaterloo/flix/language/phase/njvm/classes/Closure.scala

```diff
@@ -0,0 +1,304 @@
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimChar =>
+      val field = cg.mkPrimField[MChar]("result")
+      cg.mkMethod1[Ref[Closure], MChar]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimByte =>
+      val field = cg.mkPrimField[MByte]("result")
+      cg.mkMethod1[Ref[Closure], MByte]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimShort =>
+      val field = cg.mkPrimField[MShort]("result")
+      cg.mkMethod1[Ref[Closure], MShort]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimInt =>
+      val field = cg.mkPrimField[MInt]("result")
+      cg.mkMethod1[Ref[Closure], MInt]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimLong =>
+      val field = cg.mkPrimField[MLong]("result")
+      cg.mkMethod1[Ref[Closure], MLong]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimFloat =>
+      val field = cg.mkPrimField[MFloat]("result")
+      cg.mkMethod1[Ref[Closure], MFloat]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case PrimDouble =>
+      val field = cg.mkPrimField[MDouble]("result")
+      cg.mkMethod1[Ref[Closure], MDouble]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case Reference(_) =>
+      val field = cg.mkField[Ref[MObject]]("result")
+      cg.mkMethod1[Ref[Closure], Ref[MObject]]("getResult",
+        sig =>
+          sig.getArg1.LOAD[StackNil] |>>
+            field.GET_FIELD |>>
+            RETURN
+      )
+    case _ => throw InternalCompilerException(s"Unexpected type $returnType")
+  }
+
+  private def getResultField[T1 <: MnemonicsTypes : TypeTag]: Field[T1] =
+    new Field[T1]("result")
+
+  private def getResultPrimField[T1 <: MnemonicsPrimTypes : TypeTag] : PrimField[T1] =
+    new PrimField[T1]("result")
+
+  private def getResultMethod[T1 <: MnemonicsPrimTypes : TypeTag] : Method1[Ref[Closure], T1] =
+    new Method1(JvmModifier.InvokeVirtual,ct, "getResult")
+
+  val defaultConstructor : UncheckedVoidMethod = {
+    val setCloFields =
+      (sig : UncheckedFunSig)  =>{
+        var ins = NO_OP[StackNil]
+        var ind: Int = 2
+        for (arg <- cloFieldsTypes) {
+          ins = ins |>>
+            sig.getArg[Ref[Closure]](0).LOAD |>>
+            (arg match {
+              case PrimBool => sig.getArg[MBool](ind).LOAD[StackNil ** Ref[Closure]] |>>
```

Comment:
new line after =>, so its easier to see that there are two instructions. Same applies for the other cases.